File: UI.CSharp/ViewModels/RelayCommand.cs
using System;
using System.Windows.Input;

namespace UI.ViewModels {

    public class RelayCommand : ICommand {

        private Predicate<object> CanExecute { get; set; }
        private Action<object> Execute { get; set; }

        public RelayCommand(Predicate<object> canExecute, Action<object> execute) {
            CanExecute = canExecute;
            Execute = execute;
        }

        public event EventHandler CanExecuteChanged {
            add { CommandManager.RequerySuggested += value; }
            remove { CommandManager.RequerySuggested -= value; }
        }

        event EventHandler ICommand.CanExecuteChanged {
            add { CanExecuteChanged += value; }
            remove { CanExecuteChanged -= value; }
        }

        bool ICommand.CanExecute(object parameter) { return CanExecute(parameter); }

        void ICommand.Execute(object parameter) { Execute(parameter); }
    }
}

File: UI/Commands/RelayCommand.fs
namespace UI.Commands

open System
open System.Windows.Input

type RelayCommand(canExecute, execute) =
    interface ICommand with
        
        [<CLIEvent>]
        member x.CanExecuteChanged = CommandManager.RequerySuggested
        member x.CanExecute(parameter) = canExecute parameter
        member x.Execute(parameter) = execute parameter